sympy/series/limits.py

```diff
@@ -315,6 +367,8 @@ def set_signs(expr):
                 if ex.is_positive:
                     return S.Zero
                 elif ex == 0:
```

Comment:
`is_meromorphic` and `z0 is S.Infinity` block have this else part quite similar, is it possible to move this code to a new heuristic function (`_limit_from_leading_term`) and use that as a substitute for both? Will decrease code repetition

Reply:
I'll try this out and get back !

Reply:
Yes, I meant the entire `else` block, which includes `if ex.is_positive` and everything below

Reply:
Oh okay I got what you were trying to convey . Nope no wrong answers really , atleast I couldn't find one while running the test locally and also mathematically shouldn't make much difference . I'll make a commit sometime soon for this . Thanks

Reply:
I have implemented a basic layout for this ,You could surely review this now !